src/background/utils/extension-api.ts

```diff
@@ -64,15 +60,17 @@ export function writeSyncStorage<T extends {[key: string]: any}>(values: T): Pro
 }
 
 export function writeLocalStorage<T extends {[key: string]: any}>(values: T): Promise<void> {
-    isWriting = true;
     return new Promise<void>((resolve) => {
         chrome.storage.local.set(values, () => {
-            isWriting = false;
             resolve();
         });
     });
 }
 
+export const subscribeToOuterSettingsChange = (callback: () => void) => {
+    chrome.storage.onChanged.addListener(callback);
```

Comment:
Previous use of `isWriting` seemed to be valid. So you could use `!isWriting && ...` to ensure the callback is called for remote changes only. By the way, I don't know if `chrome.storage.onChanged` reacts on both local and outer changes.

Reply:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/onChanged Says it could be from "managed" "remote" "local"